From 03b2e8c5ffcd00f293267096c10572c72f89be27 Mon Sep 17 00:00:00 2001
From: Ashish Kalra <akalra@marvell.com>
Date: Wed, 22 May 2019 23:08:18 +0530
Subject: [PATCH 251/386] octeontx2-gti: Probing gti_watchdog driver on
 octeontx2 compatible string

Driver is probed on all platforms due to very
generic compatible string ("arm,armv8-timer").

On all Armada/CN913x following appears in the dmesg:
[    7.318707] gti wdog platform driver init
[    7.327037] gti_watchdog: probe of ap806:config-space@f0000000:timer
 failed with error -22

To resolve issues reported on Armada, change probing the driver on a
proprietary octeontx2 compatible string.

Change-Id: Ic2ab9927593cb7be7e7c0a34463dccdc4117d283
Signed-off-by: Ashish Kalra <akalra@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/9782
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Linu Cherian <lcherian@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/soc/marvell/gti/gti_watchdog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/gti/gti_watchdog.c b/drivers/soc/marvell/gti/gti_watchdog.c
index 6300e019f440..9ac6a1ae7b39 100644
--- a/drivers/soc/marvell/gti/gti_watchdog.c
+++ b/drivers/soc/marvell/gti/gti_watchdog.c
@@ -218,7 +218,7 @@ static int gti_wdog_remove(struct platform_device *pdev)
 }
 
 static const struct of_device_id gti_wdog_of_ids[] = {
-	{ .compatible = "arm,armv8-timer"},
+	{ .compatible = "marvell,octeontx2-timer"},
 	{},
 };
 
-- 
2.17.1

